packages/launcher/src/renderer/launcher/wallets/WalletsView.js

```diff
@@ -120,7 +202,28 @@ const setDefaultWalletMutation = graphql`
 class WalletsView extends Component<Props, State> {
   static contextType = EnvironmentContext
 
-  state = {}
+  constructor(props: Props) {
+    super(props)
+    this.state = { wallets: this.getWalletsArray(props) }
```

Comment:
Better use [React's `getDerivedStateFromProps()`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) for such logic